Gemfile

```diff
@@ -1,31 +1,24 @@
-# frozen_string_literal: true
-
 source "https://rubygems.org"
 
-branch = ENV.fetch('SOLIDUS_BRANCH', 'master')
+branch = ENV.fetch("SOLIDUS_BRANCH", "master")
 gem "solidus", github: "solidusio/solidus", branch: branch
 
-if branch == 'master' || branch >= "v2.0"
+if branch == "master" || branch >= "v2.3"
+  gem "rails", "~> 5.1.0" # hack for broken bundler dependency resolution
+  gem "rails-controller-testing", group: :test
+elsif branch >= "v2.0"
+  gem "rails", "~> 5.0.0" # hack for broken bundler dependency resolution
   gem "rails-controller-testing", group: :test
 else
-  gem "rails", '~> 4.2.0'
+  gem "rails", "~> 4.2.7"
   gem "rails_test_params_backport", group: :test
 end
 
-if ENV['DB'] == 'mysql'
-  gem 'mysql2', '~> 0.4.10'
-else
-  gem 'pg', '~> 0.21'
-end
+gem "pg", "~> 0.21"
```

Comment:
Do we prefer this to checking the environment variable?

Reply:
my bad, I forgot to add it again, done